src/main/java/org/jabref/preferences/WorkspacePreferences.java

```diff
@@ -21,6 +24,7 @@ public class WorkspacePreferences {
     private final BooleanProperty showAdvancedHints;
     private final BooleanProperty warnAboutDuplicatesInInspection;
     private final BooleanProperty confirmDelete;
+    private final ObjectProperty<Set<String>> selectedSlrFetchers;
```

Comment:
You will only be notified here by a listener, if the whole set is replaced.
Look into SidePanePreferences on how to deal with Sets in the preferences.